src/otx/core/data/dataset/classification.py

```diff
@@ -111,6 +111,9 @@ def _get_item_impl(self, index: int) -> TorchDataItem | None:
 
     def _convert_to_onehot(self, labels: torch.tensor, ignored_labels: list[int]) -> torch.tensor:
         """Convert label to one-hot vector format."""
+        # Torch's one_hot() expects the input to be of type long
+        # However, when labels are empty, they are of type float32
+        labels = labels.long()
         onehot = functional.one_hot(labels, self.num_classes).sum(0).clamp_max_(1)
```

Comment:
not sure if the comments are necessary. 
```suggestion
        onehot = functional.one_hot(labels.long(), self.num_classes).sum(0).clamp_max_(1)
```